cfg/builder/builder_walk.cc

```diff
@@ -327,7 +367,8 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
                 } else if (auto lhsLocal = ast::cast_tree<ast::Local>(a.lhs)) {
                     lhs = cctx.inWhat.enterLocal(lhsLocal->localVariable);
                 } else if (auto ident = ast::cast_tree<ast::UnresolvedIdent>(a.lhs)) {
-                    lhs = unresolvedIdent2Local(cctx, *ident);
+                    auto assign = true;
+                    lhs = unresolvedIdent2Local(cctx, *ident, assign);
```

Comment:
I added the `assign` parameter to avoid reporting two errors in the case the ident is indeed unresolved:

```
test.rb:49: The instance variable @bar must be declared inside initialize or declared nilable https://srb.help/5005
    49 |    @bar = T.let(1, Integer)
            ^^^^

test.rb:49: Use of undeclared variable @bar https://srb.help/6002
    49 |    @bar = T.let(1, Integer)
            ^^^^
```

Reply:
I don't really understand the cases in which `assign` is necessary: avoiding reporting errors in the case where we can't resolve the variable seems like a bad thing (we should be telling the user about issues?).  What are the particular mechanics of this change that require adding `assign`?